addons/mail/models/mail_activity.py

```diff
@@ -631,6 +643,13 @@ def _search_activity_user_id(self, operator, operand):
     def _search_activity_type_id(self, operator, operand):
         return [('activity_ids.activity_type_id', operator, operand)]
 
+    @api.multi
+    def set_exception_activity_type(self):
+        self.ensure_one()
+        self.exception_activity_type = False
```

Comment:
- This is a first write that can be useless
- writing on exception_activity_type in a loop. 
- will keep the last activity_type_id, why the last? We could break on first, or keep the more important one
- "for record" can be "for activity" to improve readability. 
Improvement suggestion:
``` python3
decoration_type = None
icon = None
for activity in self.activity_ids.filte....
    if not decoration and activity.activity_type_id.decoration_type == 'warning'
        decoration_type = 'warning'
        icon = record.activity_type_id.icon
    elif activity.activity_type_id.decoration_type == 'danger'
        decoration_type = 'warning'
        icon = record.activity_type_id.icon
        break
self.exception_activity_type = "pull-right text-%s fa %s" % (decoration_type, icon)
```
What do you think?